### PR TITLE
Compile warnings 2

### PR DESCRIPTION
**Pull request by shailesh33**
o Defines new types for usec_t and msec_t
o comment out unused functions
o type conversion errors

### CODE REVIEW

File: src/dyn_core.h [hunk 1]
```diff
@@ -234,7 +234,7 @@ struct server {
     uint32_t           ns_conn_q;     /* # server connection */
     struct conn_tqh    s_conn_q;      /* server connection q */
 
-    int64_t            next_retry;    /* next retry time in usec */
+    msec_t             next_retry;    /* next retry time in usec */
```
Comment (ipapapa) [inline src/dyn_core.h]: please update the comment to mention "msec"

File: src/dyn_dnode_msg.c [hunk 2]
```diff
@@ -1032,10 +1034,10 @@ dmsg_parse(struct dmsg *dmsg)
       rnode->is_seed = false;
 
       ts[ts_len] = '\0';
-      rnode->ts = atol(ts);
+      rnode->ts = (uint64_t)atol((char*)ts);
```
Comment (ipapapa) [inline src/dyn_dnode_msg.c]: the atol(ts) was providing a warning? It was looking correct to me.

File: src/dyn_server.c [hunk 3]
```diff
@@ -275,19 +275,19 @@ server_failure(struct context *ctx, struct server *server)
 		return;
 	}
 
-	now = dn_usec_now();
-	if (now < 0) {
+	now = dn_msec_now();
+	if (now == 0) {
 		return;
```
Comment (shailesh33) [inline src/dyn_server.c]: it returns 0.

File: src/dyn_dnode_msg.c (hunk 2 above)
Comment (ipapapa) [inline src/dyn_dnode_msg.c]: the warning is about the (uint64_t) not about the ((char*)

Comment (shailesh33) [inline src/dyn_dnode_msg.c]: they both complain.

File: src/dyn_server.c (hunk 3 above)
Comment (ipapapa) [inline src/dyn_server.c]: https://github.com/Netflix/dynomite/blob/dev/src/dyn_util.c#L459

Comment (shailesh33) [inline src/dyn_server.c]: oh got it. I changed the function to return 0. The change is in this PR itself

**Comment by shailesh33** [comment]
I messed up the commits here. I dont want people to laugh at our commit history. created compile-warnings-3

**Pull request closed without merging**